src/main/java/frc/robot/utils/MotorMovement.java

```diff
@@ -0,0 +1,29 @@
+package frc.robot.utils;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxLimitSwitch.Type;
+
+public final class MotorMovement {
+    private MotorMovement() {
+    }
+
+    public static boolean isMovementLimited(CANSparkMax controller, double setpoint, Type limitType) {
+        return
+            (setpoint > 0 && controller.getForwardLimitSwitch(limitType).isPressed()) ||
+            (setpoint < 0 && controller.getReverseLimitSwitch(limitType).isPressed());
+    }
+
+    public static boolean isOutOfBounds(CANSparkMax controller, Type limitType) {
+        return
+            controller.getForwardLimitSwitch(limitType).isPressed() ||
+            controller.getReverseLimitSwitch(limitType).isPressed();
+    }
```

Comment:
We don't have limit switches this year. Instead, we're using the REV absolute encoders and the NEO built-in encoders (we need both since the absolute encoders have very coarse accuracy, whereas the built-in hall effect sensors are very accurate). You'll need to add some method of fusing the two sensors together, using the absolute encoder for zeroing and the built-in encoder for actual fine measurement.